### DISCUSSION TITLE
Updating collection with unique constraints causes "duplicate key value violates unique constraint "

### DISCUSSION THREAD

**Discussion by jnsvu** [mikro-orm/mikro-orm#2832] [Q&A]
**Describe the bug**
I have the following entities:
```
@Entity()
class User {
  @PrimaryKey()
  id!: number

 @OneToMany(() => Schedule, (schedule) => schedule.user, { eager: true })
  schedule = new Collection<Schedule>(this)

}

@Entity()
@Unique({ properties: ["user", "weekday"] })
class Schedule {
  @PrimaryKey()
  id!: number

  @ManyToOne(() => User, { hidden: true })
  user!: User

  @Enum()
  weekday!: Weekday
}
```

When trying to update the schedule like so:
```
const newSchedule = [{
weekday: Weekday.Monday,
}]

wrap(user).assign({
 schedule: newSchedule
} { updateNestedEntities: true })

await DI.em.flush()

```
I get unique constraint error: 
`duplicate key value violates unique constraint "schedule_user_id_weekday_unique"`

It works if I remove the unique constraint from `Schedule`.

Am I doing something wrong here or is this a bug / supposed to happen?

Probably related to #2273 fix was fixed in v5 

**Versions**

| Dependency | Version |
| - | - |
| node | 14 |
| typescript | 4.5 |
| mikro-orm | 5.0.3 |
| your-driver | postgresql |

**Comment by B4nan** [comment]
Please always upgrade to latest before reporting issues. Note that `updateNestedEntities` is enabled by default in v5. The issue you linked seems unrelated, here you are explicitly creating unique constraint that throws - so you already have such pair in your database - in other words, this does not seem like a bug in the ORM at all. Moving to discussion, we can create issue if we actually have a failing reproduction - but given this is a db constraint check that throws, it literally means you have problem in your data. Your assign call won't create new schedule, it will trigger an update query. It is up to you to make sure that update query makes sense.

**Comment by productdevbook** [comment]
@sniib  Did you solve this problem? I faced similar problem.
